src/stream.iced

```diff
@@ -0,0 +1,114 @@
+stream = require('keybase-chunk-stream')
+nacl = require('keybase-nacl')
+armor = require('node-armor-x')
+msgpack = require('msgpack-lite')
+payload = require('./payload')
+header = require('./header')
+nonce = require('./nonce')
+format = require('./format')
+
+saltpack_block_len = (1024**2)
+noop = () ->
+
+class NaClEncryptStream extends stream.ChunkStream
+  _encrypt : (chunk) =>
+    payload_list = payload.generate_encryption_payload_packet(@_encryptor, chunk, @_block_num, @_header_hash, @_mac_keys)
+    ++@_block_num
+    return payload_list
+
+  _transform : (chunk, encoding, cb) ->
+    if not @_header_written
+      {header_intermediate, header_hash, mac_keys, payload_key} = header.generate_encryption_header_packet(@_encryptor, @_recipients, {anonymized_recipients : @_anonymized_recipients})
+      @_header_hash = header_hash
+      @_mac_keys = mac_keys
+      @_encryptor.secretKey = payload_key
+      @push(header_intermediate)
+      @_header_written = true
+    super(chunk, encoding, cb)
```

Comment:
Is this relying on the fact that we passed the `@_encrypt` method as the first argument to the `super` constructor below? It's a little bit hard to follow without comments.